File: app/src/main/kotlin/com/example/unittesting/login/model/LoginRepository.java
package com.example.unittesting.login.model;

import java.util.concurrent.TimeUnit;

import io.reactivex.Observable;
import timber.log.Timber;

public class LoginRepository {

    static final String CORRECT_LOGIN = "dbacinski";
    static final String CORRECT_PASSWORD = "correct";

    public Observable<Boolean> login(String login, String password) {
        Timber.v("login %s with password %s", login, password);

        return Observable.just(
                CORRECT_LOGIN.equals(login) && CORRECT_PASSWORD.equals(password)
        ).delay(500, TimeUnit.MILLISECONDS);
    }
}

File: app/src/main/kotlin/com/example/unittesting/login/model/LoginRepository.kt
package com.example.unittesting.login.model

import io.reactivex.Observable
import io.reactivex.Observable.fromCallable
import timber.log.Timber
import java.util.concurrent.TimeUnit

class LoginRepository {

    fun login(login: String, password: String): Observable<Boolean> {
        Timber.v("login %s with password %s", login, password)

        return fromCallable { CORRECT_LOGIN == login && CORRECT_PASSWORD == password }
                .delay(2000, TimeUnit.MILLISECONDS)
    }

    companion object {
        internal val CORRECT_LOGIN = "dbacinski"
        internal val CORRECT_PASSWORD = "correct"
    }
}
